secret-manager/runtime/src/main/java/io/quarkiverse/googlecloudservices/secretmanager/runtime/config/SecretManagerConfigSource.java

```diff
@@ -36,14 +58,43 @@ public Set<String> getPropertyNames() {
     }
 
     @Override
-    public String getValue(String propertyName) {
-        SecretVersionName secretVersionName = SecretManagerConfigUtils.getSecretVersionName(propertyName, defaultProject);
-        if (secretVersionName == null) {
-            // The propertyName is not in the form "${sm//...}" so return null.
-            return null;
+    public Map<String, String> getProperties() {
+        return Collections.emptyMap();
+    }
+
+    void closeClient() {
+        closed.compareAndSet(false, true);
+        client.close();
+    }
+
+    private static SecretManagerServiceClient createClient(
+            final GcpBootstrapConfiguration gcpConfig,
+            final String projectId) {
+
+        try {
+            return SecretManagerServiceClient.create(
+                    SecretManagerServiceSettings.newBuilder()
+                            .setQuotaProjectId(projectId)
+                            .setCredentialsProvider(() -> credentials(gcpConfig))
+                            .build());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
         }
+    }
+
+    private static final String CLOUD_OAUTH_SCOPE = "https://www.googleapis.com/auth/cloud-platform";
 
-        AccessSecretVersionResponse response = clientProvider.get().accessSecretVersion(secretVersionName);
-        return response.getPayload().getData().toStringUtf8();
+    private static GoogleCredentials credentials(final GcpBootstrapConfiguration gcpConfig) throws IOException {
+        if (gcpConfig.serviceAccountLocation().isPresent()) {
+            try (FileInputStream is = new FileInputStream(gcpConfig.serviceAccountLocation().get())) {
+                return GoogleCredentials.fromStream(is).createScoped(CLOUD_OAUTH_SCOPE);
+            }
+        } else if (gcpConfig.serviceAccountEncodedKey().isPresent()) {
+            byte[] decode = Base64.getDecoder().decode(gcpConfig.serviceAccountEncodedKey().get());
+            try (ByteArrayInputStream is = new ByteArrayInputStream(decode)) {
+                return GoogleCredentials.fromStream(is).createScoped(CLOUD_OAUTH_SCOPE);
+            }
+        }
+        return GoogleCredentials.getApplicationDefault().createScoped(CLOUD_OAUTH_SCOPE);
```

Comment:
This is duplicated with the code in `GcpCredentialProducer.googleCredential()` but as I understand we cannot use a CDI bean here. Maybe the best will be to make the creation of the `GoogleCredentials` from a  `GcpBootstrapConfiguration` in a static method inside `GcpCredentialProducer` so it can be used in both places.

Reply:
Yes, I've considered that option.

The issue is that `GcpCredentialProducer.googleCredential()` has additional CDI dependencies, namely token authentication, which uses the OIDC extension (if available):

https://github.com/quarkiverse/quarkus-google-cloud-services/blob/38a890522af6cd08410cad08f11064030a27a595/common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/GcpCredentialProducer.java#L41-L67

I had to remove that piece for the ConfigSource part, because we cannot support it.

Reply:
Oh yes, I read the code too quickly.
OK then, I'll rebease on main to have a fresh CI build and if everything is green I'll merge it.
Thanks